### PR TITLE
WIP Feature/pdb state controller

### PR DESCRIPTION
**Pull request by TimBuckers**

### CODE REVIEW

File: march_hardware_interface/config/testsetup/controllers.yaml [hunk 1]
```diff
@@ -6,6 +6,9 @@ march:
     temperature_sensor:
       type: march_temperature_sensor_controller/MarchTemperatureSensorController
       publish_rate: 50
+    pdb_state:
+      type: march_pdb_state_controller/MarchPdbStateController
+      publish_rate: 50
```
Comment (Ishadijcks) [inline march_hardware_interface/config/testsetup/controllers.yaml:11]: Do we want to launch this for the testsetup?

File: march_hardware_interface/include/march_hardware_interface/march_hardware_interface.h [hunk 2]
```diff
@@ -13,6 +13,7 @@
 #include <ros/ros.h>
 #include <march_hardware_interface/march_hardware.h>
 #include <march_hardware_interface/march_temperature_sensor_interface.h>
+#include <march_hardware_interface/march_pdb_state_interface.h>
```
Comment (Ishadijcks) [inline march_hardware_interface/include/march_hardware_interface/march_hardware_interface.h:16]: Shouldn't be needed as it is included via the `march_hardware.h` in line 14, same goes for the `march_temperature_sensor_interface.h`

File: march_hardware_interface/include/march_hardware_interface/march_pdb_state_interface.h [hunk 3]
```diff
@@ -0,0 +1,44 @@
+// Copyright 2019 Project MarchMarchPdbStateHandlen PDO Map (total bits 272, only 2
```
Comment (Ishadijcks) [inline march_hardware_interface/include/march_hardware_interface/march_pdb_state_interface.h]: Copy paste error

Comment (Ishadijcks) [inline march_hardware_interface/include/march_hardware_interface/march_pdb_state_interface.h]:
```suggestion
// Copyright 2019 Project March.
```

File: march_hardware_interface/include/march_hardware_interface/march_pdb_state_interface.h [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+#include <string>
+
+namespace march_hardware_interface
+{
+class MarchPdbStateHandle
+{
+public:
+  MarchPdbStateHandle(
+      const std::string& name,      ///< The name of joint
+      const double* pdb_current    ///< A pointer to the storage of the pdb current value in ??.
```
Comment (Ishadijcks) [inline march_hardware_interface/include/march_hardware_interface/march_pdb_state_interface.h:16]: `??` ?

File: march_hardware_interface/launch/march3_hardware.launch [hunk 5]
```diff
@@ -10,6 +10,7 @@
           args="
                 /march/controller/state
```
Comment (Ishadijcks) [inline march_hardware_interface/launch/march3_hardware.launch:11]: Maybe we should rename this one to `joint_state`

File: march_pdb_state_controller/.gitignore [hunk 6]
```diff
@@ -0,0 +1,3 @@
+bin
+build
+lib
```
Comment (Ishadijcks) [inline march_pdb_state_controller/.gitignore:3]: Is this needed?

**Review by Ishadijcks — COMMENTED** [review]
Partial review

File: march_hardware_interface/config/testsetup/controllers.yaml (hunk 1 above)
Comment (TimBuckers) [reply march_hardware_interface/config/testsetup/controllers.yaml:11]: Depending on which testsetup launch file you pick pdb can be launched

**Pull request closed without merging**